components/automate-dex/habitat/templates/config.yml

```diff
@@ -223,6 +219,15 @@ connectors:
       nameIDPolicyFormat: {{cfg.connectors.saml.name_id_policy_format}}
   {{- end}}
 {{- end}}
+oauth2:
+  responseTypes:
+    - code
+    - token
+    - id_token
+  skipApprovalScreen: true
+{{- if cfg.connectors.saml}}
+  alwaysShowLoginScreen: true
+{{- end}}
```

Comment:
I am not clear on what is the approval screen vs the login screen, or the whole process flow here for that matter. 🤷 You state:
> To not have users click "sign in with X" if they only have LDAP, or only local
> users, enabled, but have them go straight to the username/password field, we
> set alwaysShowLoginScreen to true only if SAML is in the mix.

Am I correct, for instance, that SAML is different than local or LDAP, in that we do not ask for the credentials, rather the SAML IdP does. So if we remove the "Sign in with SAML" (that, I guess, is on the login screen?) that is why one could never log out, yes?

Reply:
> Am I correct, for instance, that SAML is different than local or LDAP, in that we do not ask for the credentials, rather the SAML IdP does. 

Yes, the "SAML sign in page" is a redirect to the IdP, no username/password prompt.

> So if we remove the "Sign in with SAML" (that, I guess, is on the login screen?) that is why one could never log out, yes?

Signing out of A2 will get you back to dex. If dex decided that, since there's only one sign in method, that must be the one you want to use, it sends you to that method's sign in page. For SAML, that's a redirect to your IdP. If you still have an active session with the IdP (that's something we don't control), you'll be signed in automatically, and redirect to A2. So, from a user's perspective, there's no way to sign out. Clicking on "Sign out" gets you signed in again. (I'm just rephrasing, I think your understanding is correct.)

Reply:
The approval screen is this:
![image](https://user-images.githubusercontent.com/870638/95172696-fa9dfe80-07b7-11eb-88fa-bd0f457f433b.png)

We have never enabled that, since it doesn't make sense in our setup.

Reply:
Thanks!